File: src/main/java/ir/syrent/wanted/Events/PlayerJoinListener.java
package ir.syrent.wanted.Events;

import ir.syrent.wanted.Utils.SkullBuilder;
import ir.syrent.wanted.WantedManager;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;

public class PlayerJoinListener implements Listener {

    @EventHandler
    public void onPlayerJoin(PlayerJoinEvent event) {
        Player player = event.getPlayer();
        if (WantedManager.getInstance().getWanted(player.getName()) != 0) {
            SkullBuilder.getInstance().cache.remove(player);
        }
    }
}

File: src/main/kotlin/ir/syrent/wanted/listener/PlayerJoinListener.kt
package ir.syrent.wanted.listener

import ir.syrent.wanted.WPlayer
import ir.syrent.wanted.core.WantedManager
import ir.syrent.wanted.storage.Database
import ir.syrent.wanted.storage.Settings
import me.mohamad82.ruom.Ruom
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent

class PlayerJoinListener: Listener {

    init {
        Ruom.registerListener(this)
    }

    @EventHandler
    fun onPlayerJoin(event: PlayerJoinEvent) {
        val player = event.player

        if (!WantedManager.wPlayers.containsKey(player.uniqueId)) {
            Database.instance.getWPlayer(player.uniqueId).whenComplete { wPlayer, _ ->
                WantedManager.wPlayers[player.uniqueId] = wPlayer ?: WPlayer(player.uniqueId, player.name, Settings.defaultWanted)
            }
        }
    }
}